src/controllers/api/2014/spellController.ts

```diff
@@ -1,6 +1,25 @@
 import { Request, Response, NextFunction } from 'express';
+import { z } from 'zod';
 import { ResourceList, escapeRegExp, redisClient } from '@/util';
 
+// --- Zod Schemas ---
+const IndexQuerySchema = z.object({
+  name: z.string().optional(),
+  level: z
+    .union([z.string().regex(/^\d+$/), z.array(z.string().regex(/^\d+$/))])
+    .optional()
+    .transform((val) => (val ? (Array.isArray(val) ? val : [val]) : undefined)),
```

Comment:
we do this in a few places, maybe a shared util fn
```ts
function normalizeToArray<T>(val: T | T[] | undefined) {
  if(!val) return undefined;
  return Array.isArray(val) ? val : [val];
}
```

applies to files before this one too.

Reply:
Done with the consolidated file.